### PR TITLE
feat(cluster.tf): add support for setting cgroup mode

### LINKED ISSUE #1838: support cgroup mode
### TL;DR

Hi, would it be possible to have [cgroup mode](https://github.com/hashicorp/terraform-provider-google/pull/16103) supported in the next version of this module?

### Terraform Resources

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#cgroup_mode
https://github.com/hashicorp/terraform-provider-google/blob/HEAD/CHANGELOG.md#510-Oct-9-2023

### Additional information

I know this mode was introduced in v5.1 of the provider while the current version of the module (v29) is based on v5.0 on the provider.

Not sure how the development of this module is working, will it be embedded "automatically" in the next version (v30) since it will be based at least on v5.1 of the provider? Or each new feature need to be selected to be in the future version of this module?

### PR DESCRIPTION
**Pull request by DrFaust92**
Fix #1838
Supersedes #1891

seeing if i can push it over the finish line

### CODE REVIEW

**Comment by apeabody** [comment]
From INT:

```
Error: expected node_config.0.linux_node_config.0.cgroup_mode to be one of ["CGROUP_MODE_UNSPECIFIED" "CGROUP_MODE_V1" "CGROUP_MODE_V2"], got 

  with module.example.module.gke.google_container_node_pool.pools["pool-03"],
  on ../../../modules/beta-public-cluster/cluster.tf line 637, in resource "google_container_node_pool" "pools":
 637:   node_config {
```

File: autogen/main/variables_defaults.tf.tmpl [hunk 1]
```diff
@@ -114,5 +114,14 @@ locals {
     ),
     var.node_pools_linux_node_configs_sysctls
   )
+   node_pools_cgroup_mode = merge(
+    { all = "" },
```
Comment (DrFaust92) [inline autogen/main/variables_defaults.tf.tmpl:118]: @apeabody i wonder if this needs to be `CGROUP_MODE_UNSPECIFIED` to make it work

Comment (apeabody) [reply autogen/main/variables_defaults.tf.tmpl:118]: Hi @DrFaust92 - I checked the provider code and `cgroup_mode` is an (optional) `string`, but will not accept `""`.  So an option for https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/2001/files#diff-a3a07be3819af6f1a65ed243a613e69b544b7274d248a4b4d7d4d127aec40aeeR1045 would be to check the locals value and if `""` instead pass 'null'.

**Comment by DrFaust92** [comment]
Thanks! Will take a look

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @DrFaust92!

Confirm `cgroup_mode` in TPG v5.25.0

**Pull request merged**